### PR TITLE
Update Google Cloud BigQuery to Akka 2.6

### PR DESCRIPTION
**Pull request by seglo**
References #2476

### CODE REVIEW

File: google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala [hunk 1]
```diff
@@ -26,12 +27,12 @@ object BigQueryAttributes {
   /**
    * Resolves the most specific [[BigQuerySettings]] for some [[Attributes]]
    */
-  def resolveSettings(attr: Attributes, mat: Materializer): BigQuerySettings =
```
Comment (seglo) [inline google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala:29]: @armanbilge Not sure if this was intended to be a public method, if not we can roll this back and make it internal.

Comment (armanbilge) [reply google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala:29]: It was intended to be public—access to these settings/attributes is public anyway, so I would think only helps to have this utility method.

**Review by armanbilge — COMMENTED** [review]
I'm a little confused why the use of a `Materializer` in the method signature is anti-2.6? The signature of `resolveSettings` was intended to match `fromMaterializer`, since that's where it's used.

File: google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala (hunk 1 above)
Comment (seglo) [reply google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala:29]: I see its convenience when called from `fromMaterializer` but I'm not sure when this method be useful to an end user.

Comment (armanbilge) [reply google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala:29]: It's useful whenever a `Future`-returning method is used from within a Stream operator. Here's an example from our code base:
```scala
Sink.fromMaterializer { (attr, mat) =>
  implicit val settings = BigQueryAttributes.resolveSettings(attr, mat)
  Sink.lazyFutureSink { () =>
    for {
      table <- BigQuery.createTable[A](datasetId, tableId) // Uses implicit settings
    } yield BigQuery.insertAll(datasetId, tableId, ...)
  }
}
```

Unless the problem is the use of `Future`-returning methods in the BigQuery API. I've seen some of the other Alpakka APIs prefer the use of `Source` everywhere even if returning a single item.

Comment (seglo) [reply google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala:29]: I see. In the main stream you settings would be added as an attribute so they can be accessed like this for stages you build separately from each other. 

It wasn't immediately obvious to me that settings would be reused in more than one stage, but it makes sense now.

Comment (armanbilge) [reply google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala:29]: This is why I was asking about the need for the signature change—I know it's minor but `resolveSettings(attr, mat)` is just a little less annoying than `resolveSettings(attr, mat.system)`.

Comment (seglo) [reply google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala:29]: Yes, you've convinced me.  I'll close the PR.

Comment (armanbilge) [reply google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryAttributes.scala:29]: Thanks, appreciate that!

**Pull request closed without merging**